docs/DESIGN.md

```diff
@@ -0,0 +1,54 @@
+# Design
+
+## Vision
+
+> High-level goals defining the long term vision for the project that will guide daily activities, its design, and feature acceptance.
+
+- **High performance**: auto-instrumentation performance impact should not be a concern for its users.
+- **Reliability**: stable and performant under different loads. Well-behaved under extreme load, with predictable, low resource consumption.
+- **Visibility**: users should be able to generate telemetry data that provides deep and detailed visibility into their applications. Such telemetry must allow users to identify and solve application-related issues in production.
+- **Useful out-of-the-box**: after install users should be able to get telemetry data from targeted libraries with none or minimal configuration (good selection of defaults).
+- **Extensible**: key components can be chosen either at build time.
+
+## Constraints
+
+- No compliance to Common Language Specification (CLS). More: [#131](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/131).
```

Comment:
Perhaps this is also a good place to add the library dependency restrictions.Maybe something like, `No dependencies to external libraries to prevent customer applications from breaking if they do not rely on that library or that version of the library.` And then we can include a link to some of the discussions that have occurred in PRs.

Reply:
I tried to refine the whole section here: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/137/commits/7139aaacb6976bc15f9d18014fa7ead0d936c09a

Reply:
That looks like a good change. 👍